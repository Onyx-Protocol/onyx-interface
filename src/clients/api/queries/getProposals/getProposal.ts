import { VError } from 'errors';
import { formatToProposal, restService } from 'utilities';

import { GetProposalInput, GetProposalOutput, ProposalDetailApiResponse } from './types';

const getProposal = async ({ id }: GetProposalInput): Promise<GetProposalOutput> => {
  const response = await restService<ProposalDetailApiResponse>({
    endpoint: `/proposal/${id}`,
    method: 'GET',
    gov: true,
  });

  const payload = response.data?.data;

  // @todo Add specific api error handling
  // if ('result' in response && response.result === 'error') {
  //   throw new VError({
  //     type: 'unexpected',
  //     code: 'somethingWentWrong',
  //     data: { message: response.message },
  //   });
  // }

  if (!payload) {
    throw new VError({ type: 'unexpected', code: 'somethingWentWrong' });
  }

  const convertedPayload = {
    ...payload.data,
    for_votes: payload.data.forVotes,
    against_votes: payload.data.againstVotes,
    start_block: payload.data.startBlock,
    start_tx_hash: payload.data.startTxHash,
    start_timestamp: payload.data.startTimestamp,
    cancel_block: payload.data.cancelBlock,
    cancel_tx_hash: payload.data.cancelTxHash,
    cancel_timestamp: payload.data.cancelTimestamp,
    end_block: payload.data.endBlock,
    end_tx_hash: payload.data.endTxHash,
    end_timestamp: payload.data.endTimestamp,
    voter_count: payload.data.voterCount,
    created_at: payload.data.createdAt,
    updated_at: payload.data.updatedAt,
  };

  return formatToProposal(convertedPayload);
};

export default getProposal;
