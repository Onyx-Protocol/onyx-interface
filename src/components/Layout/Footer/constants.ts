export const XCN_MEDIUM_URL = 'https://medium.com/OnyxProtocol';
export const XCN_DISCORD_URL = 'https://discord.com/invite/onyxprotocol';
export const XCN_TELEGRAM_URL = 'https://t.me/OnyxOrg';
export const XCN_TWITTER_URL = 'https://twitter.com/OnyxProtocol';
export const XCN_GITHUB_URL = 'https://GitHub.com/Onyx-Protocol';
