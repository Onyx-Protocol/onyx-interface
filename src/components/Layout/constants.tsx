import Path from 'constants/path';

import { MenuItem } from './types';

/**
 * Updated by ozdav
 * Just commented to hide some menu links
 */
export const menuItems: MenuItem[] = [
  {
    href: Path.ROOT,
    // Translation key: do not remove this comment
    // t('layout.menuItems.staking')
    i18nKey: 'layout.menuItems.stake',
    i18nTitleKey: 'layout.menuItems.stake',
    icon: 'stake',
  },
  // {
  //   href: Path.ROOT,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.dashboard')
  //   i18nKey: 'layout.menuItems.dashboard',
  //   i18nTitleKey: 'layout.menuItems.dashboard',
  //   icon: 'dashboard',
  // },
  // {
  //   href: Path.MARKETS,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.markets')
  //   i18nKey: 'layout.menuItems.markets',
  //   i18nTitleKey: 'layout.menuItems.markets',
  //   icon: 'market',
  // },
  {
    href: Path.HISTORY,
    // Translation key: do not remove this comment
    // t('layout.menuItems.history')
    i18nKey: 'layout.menuItems.history',
    i18nTitleKey: 'layout.menuItems.history',
    icon: 'history',
  },
  {
    href: Path.SWAP,
    // Translation key: do not remove this comment
    // t('layout.menuItems.swap')
    i18nKey: 'layout.menuItems.swap',
    i18nTitleKey: 'layout.menuItems.swap',
    icon: 'convert',
    // isNew: true,
  },
  {
    href: Path.FARM,
    // Translation key: do not remove this comment
    // t('layout.menuItems.farm')
    i18nKey: 'layout.menuItems.farm',
    i18nTitleKey: 'layout.menuItems.farm',
    icon: 'farm',
    // isNew: true,
  },
  {
    href: Path.GOVERNANCE,
    // Translation key: do not remove this comment
    // t('layout.menuItems.governance')
    i18nKey: 'layout.menuItems.governance',
    i18nTitleKey: 'layout.menuItems.governance',
    icon: 'vote',
  },
  {
    href: Path.AIAGENT,
    // Translation key: do not remove this comment
    // t('layout.menuItems.governance')
    i18nKey: 'layout.menuItems.aiagent',
    i18nTitleKey: 'layout.menuItems.aiagent',
    icon: 'aiagent',
    isNew: true,
  },
  // {
  //   href: Path.XCN,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.xcn')
  //   i18nKey: 'layout.menuItems.xcn',
  //   i18nTitleKey: 'layout.menuItems.xcn',
  //   icon: 'xcnOutlined',
  // },
  // {
  //   href: Path.LIQUIDATE,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.xcn')
  //   i18nKey: 'layout.menuItems.liquidate',
  //   i18nTitleKey: 'layout.menuItems.liquidate',
  //   icon: 'liquidate',
  // },
  // {
  //   href: Path.WPUNKS,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.xcn')
  //   i18nKey: 'layout.menuItems.fpunk',
  //   i18nTitleKey: 'layout.menuItems.fpunk',
  //   icon: 'wpunksOutlined',
  // },
  // TO BE ACTIVATED ONCE CALCULATOR IS READY
  {
    href: Path.GOVERNANCE_LEADER_BOARD,
    // Translation key: do not remove this comment
    // t('layout.menuItems.xcn')
    i18nKey: 'layout.menuItems.leaderboard',
    i18nTitleKey: 'layout.menuItems.leaderboard',
    isNew: true,
    icon: 'xcnOutlined',
  },
  {
    href: Path.POINTS,
    // Translation key: do not remove this comment
    // t('layout.menuItems.points')
    i18nKey: 'layout.menuItems.points',
    i18nTitleKey: 'layout.menuItems.points',
    icon: 'gem',
    isNew: true,
    subMenu: [
      {
        href: Path.POINTS,
        // Translation key: do not remove this comment
        // t('layout.menuItems.pointsDashboard')
        i18nKey: 'layout.menuItems.pointsDashboard',
        i18nTitleKey: 'layout.menuItems.pointsDashboard',
        icon: 'dashboard',
      },
      {
        href: Path.POINTS_LEADERBOARD,
        // Translation key: do not remove this comment
        // t('layout.menuItems.pointsLeaderboard')
        i18nKey: 'layout.menuItems.pointsLeaderboard',
        i18nTitleKey: 'layout.menuItems.pointsLeaderboard',
        icon: 'person',
      },
    ],
  },
];
