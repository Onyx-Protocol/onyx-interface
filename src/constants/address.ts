export const NULL_ADDRESS = '0x0000000000000000000000000000000000000000';
export const UINT_MAX = '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff';
