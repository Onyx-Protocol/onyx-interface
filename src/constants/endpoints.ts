import { EthChainId } from 'types';

export const API_ENDPOINT_URLS = {
  [EthChainId.MAINNET]: 'https://api.onyx.org/api',
  [EthChainId.TESTNET]: 'https://testapi.onyx.org/api',
};

export const API_GOV_ENDPOINT_URLS = {
  [EthChainId.MAINNET]: 'https://govapi.onyx.org',
  [EthChainId.TESTNET]: 'https://testapi.onyx.org/gov',
};

export const RPC_URLS: {
  [key: string]: string[];
} = {
  [EthChainId.MAINNET]: ['https://mainnet.infura.io/v3/2a396817d3e349d289318db583cc123c'],
  [EthChainId.TESTNET]: ['https://goerli.infura.io/v3/2a396817d3e349d289318db583cc123c'],
};

export const SUBGRAPH_LINKS: { [key: number]: string[] } = {
  1: ['https://api.thegraph.com/subgraphs/name/seme0801/onyx-subgraph'],
  5: ['https://api.thegraph.com/subgraphs/name/seme0801/onyx-goerli-subgraph'],
};
