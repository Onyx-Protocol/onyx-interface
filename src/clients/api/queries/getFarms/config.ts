import { EthChainId } from 'types';

import { TOKENS, UNISWAP_TOKENS } from 'constants/tokens';

import { FarmConfig } from './types';

const farmConfig: FarmConfig[] = [
  {
    pid: 0,
    lpSymbol: 'XCN-WETH LP',
    lpAddresses: {
      [EthChainId.MAINNET]: '0x859f7092f56c43BB48bb46dE7119d9c799716CDF',
      [EthChainId.TESTNET]: '',
    },
    token: TOKENS.xcn,
    quoteToken: UNISWAP_TOKENS.weth,
  },

  // {
  //   pid: 2,
  //   lpSymbol: 'XCN-WETH LP',
  //   lpAddresses: {
  //     [EthChainId.MAINNET]: '',
  //     [EthChainId.TESTNET]: '0x006Df670C4735A18422B8c75F06d891D67e6B405',
  //   },
  //   token: TOKENS.xcn,
  //   quoteToken: UNISWAP_TOKENS.weth,
  // },

  // {
  //   pid: 3,
  //   lpSymbol: 'XCN-USDT LP',
  //   lpAddresses: {
  //     [EthChainId.MAINNET]: '',
  //     [EthChainId.TESTNET]: '0xe346BcA6aDFCA5E6474a43E1d73270E43e80F41e',
  //   },
  //   token: TOKENS.xcn,
  //   quoteToken: TOKENS.usdt,
  // },
];

export default farmConfig;
