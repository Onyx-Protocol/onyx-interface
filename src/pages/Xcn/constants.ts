import BigNumber from 'bignumber.js';

import { TOKENS } from 'constants/tokens';

export const MINTED_XCN = '23700000';
export const MINTED_XCN_WEI = new BigNumber(MINTED_XCN).times(
  new BigNumber(10).pow(TOKENS.xcn.decimals),
);
