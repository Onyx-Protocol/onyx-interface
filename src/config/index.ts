import sample from 'lodash/sample';
import { EthChainId } from 'types';

import { ETH_SCAN_URLS } from 'constants/ethereum';
import { API_ENDPOINT_URLS, API_GOV_ENDPOINT_URLS, RPC_URLS } from 'constants/endpoints';

export interface Config {
  chainId: EthChainId;
  isOnTestnet: boolean;
  rpcUrl: string;
  apiUrl: string;
  apiGovUrl: string;
  ethScanUrl: string;
}

const chainId: EthChainId = process.env.REACT_APP_CHAIN_ID
  ? parseInt(process.env.REACT_APP_CHAIN_ID, 10)
  : EthChainId.MAINNET;

const isOnTestnet = chainId === EthChainId.TESTNET;
const rpcUrl = sample(RPC_URLS[chainId]) as string;
const apiUrl = API_ENDPOINT_URLS[chainId];
const apiGovUrl = API_GOV_ENDPOINT_URLS[chainId];
const ethScanUrl = ETH_SCAN_URLS[chainId];

const config: Config = {
  chainId,
  isOnTestnet,
  rpcUrl,
  apiUrl,
  apiGovUrl,
  ethScanUrl,
};

export default config;
