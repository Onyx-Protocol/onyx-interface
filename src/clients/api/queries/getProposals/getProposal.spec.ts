import config from 'config';
import { enableFetchMocks } from 'jest-fetch-mock';

import proposalResponse from '__mocks__/api/proposals.json';
import { SUBGRAPH_LINKS } from 'constants/endpoints';

import getProposal from './getProposal';

enableFetchMocks();

describe('api/queries/getProposal', () => {
  test('returns proposal', async () => {
    fetchMock.mockResponseOnce(
      JSON.stringify({
        status: 200,
        data: {
          proposals: [proposalResponse[0]],
          proposalCounts: [{ count: 1 }],
        },
      }),
    );

    const response = await getProposal({
      id: 1,
    });

    expect(fetchMock).toBeCalledWith(SUBGRAPH_LINKS[config.chainId].latest, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        query: `
          query proposalsQuery {
            proposals(
              first: 5,
              skip: 0,
              orderBy: createdBlockTimestamp,
              orderDirection: desc
              where: { id: "1" }
            ) {
              id
              proposer
              targets
              values
              signatures
              callDatas
              startBlock
              endBlock
              description
              state
              eta
              forVotes
              againstVotes
              createdBlockNumber
              createdBlockTimestamp
              createdTransactionHash
              queuedBlockNumber
              queuedBlockTimestamp
              queuedTransactionHash
              executedBlockNumber
              executedBlockTimestamp
              executedTransactionHash
              canceledBlockNumber
              canceledBlockTimestamp
              canceledTransactionHash
            }
            
            proposalCounts(where: { id: "0" }) { count }
          }
        `,
      }),
    });

    expect(response).toMatchSnapshot();
  });
});
