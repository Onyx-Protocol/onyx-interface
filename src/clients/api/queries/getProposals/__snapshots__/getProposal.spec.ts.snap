// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/queries/getProposal returns proposal 1`] = `
Object {
  "actions": Array [
    Object {
      "callData": "0x000000000000000000000000000000000000000000000000000000000002a300",
      "signature": "setDelay(uint256)",
      "target": "0x08eDF0F2AF8672029eb445742B3b4072c6158DF3",
      "value": "0",
    },
    Object {
      "callData": "0x0000000000000000000000000000000000000000000000000000000000003200",
      "signature": "changeVotingPeriod(uint256)",
      "target": "0xdec2F31C3984F3440540DC78Ef21b1369d4eF767",
      "value": "0",
    },
  ],
  "againstVotesWei": "0",
  "cancelDate": undefined,
  "cancelTxHash": undefined,
  "createdDate": 2022-04-22T20:52:40.000Z,
  "createdTxHash": "0x8b70679651c2e7cdeafbee647e2b52c5006a5c11d15f996edde95d020e45bda5",
  "description": Object {
    "description": " Chain Improvement Proposal 6 (CIP-6) will enable time lock parameters in the governance smart contracts to protect the protocol, the DAO, and its users. CIP-6 adds a 2 day voting window where users can place votes and a 2 day wait window before a proposal can be executed. Therefore, this 4 day window will protect the protocol from malicious attacks and set a reasonable time period for voting.",
    "title": " CIP-6 Secure Proposal Timelock",
    "version": "v1",
  },
  "endBlock": 4636946,
  "executedDate": 2022-04-22T21:12:53.000Z,
  "executedTxHash": "0xac2cead7d0a330a5bcfcdfba1ab5c39822d4b04052c5bfd4aaecd50b8c47506d",
  "forVotesWei": "2.324562998201e+26",
  "id": 6,
  "isEnded": true,
  "isStarted": true,
  "proposer": "0xe864c9439eb042e969c59dd13d38cfdb28e16d73",
  "queuedDate": 2022-04-22T21:01:30.000Z,
  "queuedTxHash": "0x53b9aaffd88bb56f4df1667fcb7058c484e90ead70f641166fee10658b1f5098",
  "startBlock": 4636916,
  "state": "Executed",
  "totalVotesWei": "2.324562998201e+26",
}
`;
